packages/KMenu/KMenuItem.vue

```diff
@@ -0,0 +1,174 @@
+<template>
+  <div
+    :data-testid="item ? `${item.title.replace(' ', '-')}-menu-item` : 'menu-item'"
+    :test-mode="testMode"
+    :class="[isOpen && expandable ? 'title-dark' : '', {'expand-item' : expandable}]"
+    aria-labelledby="menu-item"
+    class="k-menu-item">
+    <button
```

Comment:
is there a reason not to us KButton here?

Reply:
The button here does not have any styles associated with what we have in KButton, therefore used general button element. Does that make sense ?

Reply:
Would `non-visual-button` class help here?

![image](https://user-images.githubusercontent.com/67973710/140104488-5bdd0de8-c48b-4fca-b787-ba2d07fbec69.png)

Reply:
I tried using `.non-visual-button` but it has `cursor: pointer !important;` and if the menu item is not expandable I need the cursor not to be a pointer. I will see if I can change the specificity and make it non-cursor but also: is it okay to have !important for a specific styling in KMenuItem.vue file ?

Reply:
I think in this scenario using `!important` will be ok provided it is on a custom class we don't see outside of KMenu (or inside a scoped SCSS block).

Reply:
I don't think you need to render a button in the case where it is not expandable.

Reply:
There is a named slot inside KButton that I am using to display the item title, if I use v-if for the button, I won't be able to display the title using same named slot.

Reply:
All menu items are considered clickable. Fire 'clicked' event if non-expando, fire 'toggled' event if expando.